docs/concept-workflows.md

```diff
@@ -60,70 +240,45 @@ public class SubscriptionWorkflowImpl implements SubscriptionWorkflow {
-
-Another commonly asked question is whether a worker can handle more Workflow instances than its cache size or number of threads it can support. The answer is that a Workflow, when in a blocked state, can be safely removed from a worker.
-Later it can be resurrected on a different or the same worker when the need (in the form of an external event) arises. So a single worker can handle millions of open Workflow executions, assuming it can handle the update rate.
-
-## State Recovery and Determinism
-
-The Workflow state recovery utilizes event sourcing which puts a few restrictions on how the code is written. The main restriction is that the Workflow code must be deterministic which means that it must produce exactly the same result if executed multiple times. This rules out any external API calls from the Workflow code as external calls can fail intermittently or change its output any time. That is why all communication with the external world should happen through Activities. For the same reason, Workflow code must use Temporal APIs to get current time, sleep, and create new threads.
-
-To understand the Temporal execution model as well as the recovery mechanism, watch the following webcast. The animation covering recovery starts at 15:50.
-
-<ResponsivePlayer url='https://www.youtube.com/watch?v=qce_AqCkFys' />
-
-## Id Uniqueness
-
-Workflow Id is assigned by a client when starting a Workflow. It is usually a business level Id like customer Id or order Id.
-
-Temporal guarantees that there could be only one Workflow (across all Workflow types) with a given Id open per [namespace](/docs/glossary/#namespace) at any time. An attempt to start a Workflow with the same Id is going to fail with `WorkflowExecutionAlreadyStarted` error.
-
-An attempt to start a Workflow if there is a completed Workflow with the same Id depends on a `WorkflowIdReusePolicy` option:
-
-- `AllowDuplicateFailedOnly` means that it is allowed to start a Workflow only if a previously executed Workflow with the same Id failed.
-- `AllowDuplicate` means that it is allowed to start independently of the previous Workflow completion status.
-- `RejectDuplicate` means that it is not allowed to start a Workflow execution using the same Workflow Id at all.
+Again, it is important to note that this code directly implements the business logic, and if any of the invoked operations (aka [Activities](/docs/concept-activities)) take a long time, the code is not going to change.
 
-The default is `AllowDuplicate`.
+It is completely okay to be blocked on `chargeMonthlyFee` for a day or more if the downstream processing service is down or not responding.
+In the same way it is a completely normal operation to sleep for 30 days directly inside the Workflow code.
```

Comment:
I think we want to repeat that this is possible because any infra failures are not going to affect the workflow state including threads, blocking calls, and any local and workflow variables.